src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs

```diff
@@ -0,0 +1,430 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Options;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.CodeStyle.CodeStyleHelpers;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    internal class CodeStyleOptions2
```

Comment:
All code from `CodeStyleOptions` moved as-is, with a simple find and replace of public option types with their internal counterparts.